atom/browser/api/atom_api_tray.cc

```diff
@@ -137,20 +143,30 @@ void Tray::OnDragEnded() {
   Emit("drag-end");
 }
 
+void Tray::SetImageOptional(v8::Isolate* isolate, mate::Arguments* args) {
+  mate::Handle<NativeImage> image;
+  args->GetNext(&image);
+  SetImage(isolate, image);
+}
+
 void Tray::SetImage(v8::Isolate* isolate, mate::Handle<NativeImage> image) {
 #if defined(OS_WIN)
-  tray_icon_->SetImage(image->GetHICON(GetSystemMetrics(SM_CXSMICON)));
+  tray_icon_->SetImage(
```

Comment:
With old behavior this would have thrown an argument conversion error instead of silently failing with `NULL`